packages/ruleset/src/functions/response-status-codes.js

```diff
@@ -0,0 +1,89 @@
+/**
+ * This function performs a few checks on each operation's responses field:
+ * 1. Status code 400 should be used instead of 422.
+ * 2. Status code 303 or 307 should be used instead of 302.
+ * 3. Operation responses should include at least one successful (2xx) status code.
+ * 4. Operation responses should not include status code 101 when successful (2xx) status codes are present.
+ * @param {*} operation an operation within the API definition
+ * @param {*} options unused, but needed as a placeholder due to the rule calling conventions
+ * @param {*} path the array of path segments indicating the "location" of the operation within the API definition
+ * @returns an array containing the violations found or [] if no violations
+ */
+function responseStatusCodes(operation, options, { path }) {
+  if (!operation.responses) {
+    return [];
+  }
+
+  const [statusCodes, successCodes] = getResponseCodes(operation.responses);
+
+  if (statusCodes.length) {
+    const errors = [];
+
+    // Check for discouraged status codes per IBM API Handbook.
+    for (const statusCode of statusCodes) {
+      if (statusCode === '422') {
+        errors.push({
+          message:
+            'Operation `responses` should use status code 400 instead of 422 for invalid request payloads.',
+          path: [...path, 'responses', statusCode]
+        });
+      } else if (statusCode === '302') {
+        errors.push({
+          message:
+            'Operation `responses` should use status code 303 or 307 instead of 302.',
+          path: [...path, 'responses', statusCode]
+        });
+      }
+    }
+
+    // Validate the success codes.
+    if (!successCodes.length && !('101' in operation.responses)) {
+      errors.push({
+        message:
+          'Operation `responses` should include at least one success status code (2xx).',
+        path: [...path, 'responses']
+      });
+    } else {
+      if (successCodes.length && '101' in operation.responses) {
+        errors.push({
+          message:
+            'Operation `responses` should not include status code 101 when success status codes (2xx) are present.',
+          path: [...path, 'responses', '101']
+        });
+      }
+
+      // Perform checks re: status code 204.
+      for (const statusCode of successCodes) {
+        if (statusCode !== '204' && !operation.responses[statusCode].content) {
```

Comment:
Does this not duplicate the `content-entry-provided` rule? I think we ignore some other status codes there that shouldn't have bodies as well, like `304`

Reply:
Yes, I just confirmed that this one check (i.e. non-204 response has content) is implemented by the content-entry-provided rule.  I'll remove this check form this rule.